donations/views.py

```diff
@@ -0,0 +1,90 @@
+import json
+import base64
+import requests
+import urlparse
+from django.shortcuts import render
+from django.core.urlresolvers import reverse
+from django.conf import settings
+from django.http import HttpResponse, JsonResponse
+from django.contrib.sites.models import Site
+from django.contrib.auth.models import User
+from django.views.generic import ListView
+from django.views.decorators.csrf import csrf_exempt
+from models import Donation, DonationCampaign
+from utils.mail import send_mail_template
+
+
+@csrf_exempt
+def donation_complete(request):
+    '''This view listens to a notification made from paypal when someone makes
+    a donation, it validates the data and then stores the donation.
+    '''
+    params = {'cmd': '_notify-validate'}
+    for key, value in request.POST.items():
+        params[key] =  value
```

Comment:
Careful: a POST item is actually not a regular dictionary:
https://docs.djangoproject.com/en/1.10/ref/request-response/#django.http.QueryDict

So, you should really be doing:

```python
params = request.POST.copy()
params.update({'cmd': '_notify-validate'})
```